marathon_acme/clients.py

```diff
@@ -20,17 +18,57 @@ def json_content(response):
     return d.addCallback(json.loads)
 
 
+def raise_for_status(response):
+    """
+    Raises a `requests.exceptions.HTTPError` if the response did not succeed.
+    Adapted from the Requests library:
+    https://github.com/kennethreitz/requests/blob/v2.8.1/requests/models.py#L825-L837
+    """
+    http_error_msg = ''
+
+    if 400 <= response.code < 500:
+        http_error_msg = '%s Client Error for url: %s' % (
+            response.code, uridecode(response.request.absoluteURI))
+
+    elif 500 <= response.code < 600:
+        http_error_msg = '%s Server Error for url: %s' % (
+            response.code, uridecode(response.request.absoluteURI))
+
+    if http_error_msg:
+        raise HTTPError(http_error_msg, response=response)
+
+    return response
+
+
+def _default_agent(agent=None, reactor=None, pool=None):
+    """
+    Set up a default agent if one is not provided. Use a default reactor to do
+    so, unless one is not provided. The agent will set up a default
+    (non-persistent) connection pool if one is not provided.
+    """
+    if agent is None:
```

Comment:
If we can't easily get the reactor from we agent, let's not bother trying to test that. Instead, we can test combinations of {`None`, not-`None`} for each of the agent and reactor and make sure that our outputs look reasonable.